Source/src/core/Scene.cpp

```diff
@@ -339,22 +339,26 @@ void Hachiko::Scene::FogConfig::SaveFogParams(YAML::Node& node)
 
 void Hachiko::Scene::AmbientLightOptionsMenu()
 {
-    ImGui::PushItemWidth(100.0f);
-    ImGui::Text("Ambient");
-    ImGui::DragFloat("Ambient Intensity", &ambient_light.intensity, 0.001, 0.f, 5.f);
+    ImGui::TextWrapped("Ambient");
+    Widgets::DragFloatConfig cfg;
+    cfg.speed = 0.001f;
+    cfg.min = 0.0f;
+    cfg.max = 5.0f;
+    DragFloat("Ambient Intensity", ambient_light.intensity, &cfg);
     ImGuiUtils::CompactColorPicker("Ambient Color", ambient_light.color.ptr());
-    ImGui::PopItemWidth();
 }
 
 void Hachiko::Scene::FogOptionsMenu()
 {
-    ImGui::PushItemWidth(100.0f);
-    ImGui::Text("Fog");
-    ImGui::Checkbox("Use Fog", &fog.enabled);
+    ImGui::TextWrapped("Fog");
+    Widgets::Checkbox("Use Fog", &fog.enabled);
     ImGuiUtils::CompactOpaqueColorPicker("Fog Color", fog.color.ptr());
-    ImGui::DragFloat("Global Density", &fog.global_density, 0.001, 0.f, 1.f);
-    ImGui::DragFloat("Height Falloff", &fog.height_falloff, 0.001, 0.f, 1.f);
-    ImGui::PopItemWidth();
+    Widgets::DragFloatConfig cfg;
+    cfg.speed = 0.001f;
+    cfg.min = 0.0f;
+    cfg.max = 1.0f;
+    DragFloat("Global Density", fog.global_density, &cfg);
+    DragFloat("Height Falloff", fog.height_falloff, &cfg);
```

Comment:
I think in this script changes we lost some decimals on Ambient Light and Fog.
Not sure if you did because it's not so crucial and were so tiny modifications that we can't percieve.
If other reviewers didn't notice or don't mind about them we can leave them like this.
I was just curious about if we needed them or not.

Reply:
Do you see anything weird? It's supposed to behave the same as before, same limits and same increment

Reply:
It's only that we had 0.000 to tweak and now it shows 0.00
On that two specific ones the decimals make a big change I think, but as you consider.

Reply:
thank you, I updated the format, so that the UI reflects the correct precision